packages/relay-relayer/src/relay.ts

```diff
@@ -72,268 +75,251 @@ export class RelayerRelay extends CommonRelay<RelayerRelayLifecycle> {
         times: this.lifecycle.times,
       };
       await this.relay(options);
-
     } catch (e: any) {
-      logger.error(e, super.meta('ormpipe-relay'));
+      logger.error(e, super.meta("ormpipe-relay"));
     }
   }
 
   private async relay(options: RelayerRelayOptions) {
-    logger.debug('start relayer relay', super.meta('ormpipe-relay-relayer', ['relayer']));
+    logger.debug(
+      "start relayer relay",
+      super.meta("ormpipe-relay-relayer", ["relayer"])
+    );
 
     logger.debug(
-      `query last message dispatched from ${super.targetName}`,
-      super.meta('ormpipe-relay', ['relayer:relay'])
+      `query last message dispatched from ${super.sourceName}`,
+      super.meta("ormpipe-relay", ["relayer:relay"])
     );
 
-    const lastImportedMessageRoot = await super.lifecycle.targetIndexerOracle.lastImportedMessageRoot({
-      chainId: options.sourceChainId
-    });
-    if (!lastImportedMessageRoot) {
+    const lastImportedMessageHash =
+      await super.lifecycle.targetIndexerOrmp.lastImportedMessageHash({
+        fromChainId: options.sourceChainId,
+        toChainId: options.targetChainId,
+      });
+    if (!lastImportedMessageHash) {
       logger.info(
-        'not have any imported message from %s',
-        super.targetName,
-        super.meta('ormpipe-relay-relayer', ['relayer']),
+        "not have any imported message from %s",
+        super.sourceName,
+        super.meta("ormpipe-relay-relayer", ["relayer"])
       );
       return;
     }
     const fullOptions = {
       ...options,
-      lastImportedMessageRoot,
+      lastImportedMessageHash,
       times: options.times,
     };
 
-    const sourceNextMessageAccepted = await this._lastAssignedMessageAccepted(fullOptions);
+    const sourceNextMessageAccepted = await this._lastAssignedMessageAccepted(
+      fullOptions
+    );
     if (!sourceNextMessageAccepted) {
       logger.info(
         `no new assigned message accepted`,
-        super.meta('ormpipe-relay', ['relayer'])
+        super.meta("ormpipe-relay", ["relayer"])
       );
       return;
     }
-    await this._relay(
-      fullOptions,
-      sourceNextMessageAccepted,
-    );
+    await this._relay(fullOptions, sourceNextMessageAccepted);
   }
 
-
-  private async _relay(options: RelayerRelayFullOptions, sourceNextMessageAccepted: OrmpMessageAccepted) {
+  private async _relay(
+    options: RelayerRelayFullOptions,
+    sourceNextMessageAccepted: OrmpMessageAccepted
+  ) {
     if (
-      options.sourceChainId.toString() != sourceNextMessageAccepted.message_fromChainId ||
-      options.targetChainId.toString() != sourceNextMessageAccepted.message_toChainId
+      options.sourceChainId.toString() !=
+        sourceNextMessageAccepted.messageFromChainId ||
```

Comment:
If source data from indexer checked sourceChainId and targetChainId, I think this is not required.